include/httpp/http/Connection.hpp

```diff
@@ -78,6 +78,56 @@ class Connection
         return request_;
     }
 
+
+    template <typename Callable>
+    void read(size_t size, char* buffer, Callable&& callable)
+    {
+        if (!own())
+        {
+            throw std::logic_error("Invalid connection state");
+        }
+
+        size_t offset = 0;
+
+        if (not body_buffer_.empty())
+        {
+            if (size <= body_buffer_.size())
+            {
+                std::copy(begin(body_buffer_), begin(body_buffer_) + size,
+                          buffer);
+                body_buffer_.erase(begin(body_buffer_),
```

Comment:
I do, if the body is small, then I'll potentially have the next request following the body part.
It would mess with: https://github.com/daedric/httpp/blob/7e8d8c6fa43fc6359eb1b6ffe908d206a2458fcf/src/httpp/http/Connection.cpp#L152-L156 not erasing this part.